### PR TITLE
Make most unit tests run in MySQL

### PR DESCRIPTION
**Pull request by jmthibault79**
Partial completion of GAWB-491.  Some SubmissionMonitorSpec tests are set to ignore due to deadlock issues.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: docker/install.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ if [ -e "jenkins_env.sh" ]; then
 fi
 
 # catch sbt issues separately
-sbt update && echo "sbt updated successefully."
+sbt update && echo "sbt updated successfully."
```
Comment (jmthibault79) [inline docker/install.sh:13]: missed this in the last review

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala [hunk 2]
```diff
@@ -313,8 +314,10 @@ trait EntityComponent {
         recursiveGetEntityReferenceIds(idSet, idSet)
       } flatMap { ids =>
         DBIO.sequence(ids.map { id =>
-          val sql = sql"""#$baseEntityAndAttributeSql where e."id" = ${id}""".as[EntityListResult]
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala:316]: H2-only version

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/PendingBucketDeletionComponent.scala [hunk 3]
```diff
@@ -8,14 +8,17 @@ trait PendingBucketDeletionComponent {
...
-    def bucket = column[String]("bucket", O.PrimaryKey)
+    def bucket = column[String]("bucket", O.PrimaryKey, O.Length(254))
 
-    def * = (bucket) <> (PendingBucketDeletionRecord.apply _, PendingBucketDeletionRecord.unapply)
+    def * = bucket <> (PendingBucketDeletionRecord, PendingBucketDeletionRecord.unapply)
   }
 
   object pendingBucketDeletionQuery extends TableQuery(new PendingBucketDeletionTable(_)) {
-    def save(pendingBucketDeletion: PendingBucketDeletionRecord): WriteAction[PendingBucketDeletionRecord] = {
-      pendingBucketDeletionQuery insertOrUpdate pendingBucketDeletion map(_ => pendingBucketDeletion)
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/PendingBucketDeletionComponent.scala:18]: Slick doesn't handle upserts with no non-key columns for MySQL well.  Here, "insert if it doesn't exist" is equivalent behavior.

File: src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponentSpec.scala [hunk 4]
```diff
@@ -79,7 +78,7 @@ class AttributeComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
 
   it should "insert entity reference attribute" in withEmptyTestDatabase {
     val workspaceId = UUID.randomUUID()
-    runAndWait(workspaceQuery += WorkspaceRecord("testns", "testname1", workspaceId, "bucket", new Timestamp(0), new Timestamp(0), "me", false, None))
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponentSpec.scala:82]: MySQL didn't like "zero" time

**Comment by abaumann** [comment]
I narrowed down the SubmissionMonitorSpec issues to handleStatusChange within SubmissionMonitor - we catch a database locking exception and some of the workflows are marked as Failure rather than Success

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala (hunk 2 above)
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala:316]: this should work except for the double quotes. The fancy string stuff is not required. How about
val sql = sql"""#$baseEntityAndAttributeSql where e.#${quoteIdentifier("id")} = ${id}""".as[EntityListResult]

Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala:316]: That worked!  Thanks.

**Pull request merged**